app/fl_training/flow/fed_edgeserver_flow.py

```diff
@@ -112,28 +111,27 @@ def run_decentralized_offload(server: FedDecentralizedEdgeServer, learning_rate,
         fed_logger.info("clustering")
         server.cluster(options)
 
-        fed_logger.info("getting state")
-        state = server.get_state()
-        fed_logger.info(f"STATE : {state}")
-        normalized_state = []
-        for bw in state:
-            if r < 50:
-                normalized_state.append(bw / 100_000_000)
-            else:
-                normalized_state.append(bw / 10_000_000)
-        iot_bw.append(normalized_state[0])
-        edge_bw.append(normalized_state[0])  # FIXME
+        fed_logger.info("getting neighbors bandwidth")
+        neighbors_bandwidth = server.get_neighbors_bandwidth()
+        fed_logger.info(f"NEIGHBORS BANDWIDTH : {neighbors_bandwidth}")
+        neighbors_bandwidth_by_type: dict[NodeType, list[float]] = {}
+        for neighbor, bw in neighbors_bandwidth.items():
+            neighbor_type = HTTPCommunicator.get_node_type(neighbor)
+            if neighbor_type not in neighbors_bandwidth_by_type:
+                neighbors_bandwidth_by_type[neighbor_type] = []
+            neighbors_bandwidth_by_type[neighbor_type].append(bw.bandwidth)
+        client_bw.append(
+            sum(neighbors_bandwidth_by_type[NodeType.CLIENT]) / len(neighbors_bandwidth_by_type[NodeType.CLIENT]))
+        edge_bw.append(
+            sum(neighbors_bandwidth_by_type[NodeType.EDGE]) / len(neighbors_bandwidth_by_type[NodeType.EDGE]))
 
         fed_logger.info("splitting")
-        server.split(normalized_state, options)
+        server.split(neighbors_bandwidth_by_type[NodeType.CLIENT], options)
         fed_logger.info(f"Split Config : {server.split_layers}")
         server.scatter_split_layers()
 
-        if r > 49:
-            learning_rate = config.LR * 0.1
-
         fed_logger.info("initializing server")
-        server.initialize(learning_rate)
+        # server.initialize(learning_rate * (0.1 ** (r // 10)))
```

Comment:
I'm also ignoring calling the initialize method. Check if it's the desired behavior.